cli/azd/pkg/devcenter/environment_store.go

```diff
@@ -0,0 +1,212 @@
+package devcenter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/contracts"
+	"github.com/azure/azure-dev/cli/azd/pkg/devcentersdk"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+)
+
+const (
+	ConfigPath                                 = "platform.config"
+	RemoteKindDevCenter environment.RemoteKind = "devcenter"
+)
+
+// EnvironmentStore is a remote environment data store for devcenter environments
+type EnvironmentStore struct {
+	config          *Config
+	devCenterClient devcentersdk.DevCenterClient
+	prompter        *Prompter
+	manager         Manager
+}
+
+// NewEnvironmentStore creates a new devcenter environment store
+func NewEnvironmentStore(
+	config *Config,
+	devCenterClient devcentersdk.DevCenterClient,
+	prompter *Prompter,
+	manager Manager,
+) environment.RemoteDataStore {
+	return &EnvironmentStore{
+		config:          config,
+		devCenterClient: devCenterClient,
+		prompter:        prompter,
+		manager:         manager,
+	}
+}
+
+// EnvPath returns the path for the environment
+func (s *EnvironmentStore) EnvPath(env *environment.Environment) string {
+	return fmt.Sprintf("projects/%s/users/me/environments/%s", s.config.Project, env.GetEnvName())
+}
+
+// ConfigPath returns the path for the environment configuration
+func (s *EnvironmentStore) ConfigPath(env *environment.Environment) string {
+	return ""
+}
+
+// List returns a list of environments for the devcenter configuration
+func (s *EnvironmentStore) List(ctx context.Context) ([]*contracts.EnvListEnvironment, error) {
+	// If we don't have a valid devcenter configuration yet
+	// then prompt the user to initialize the correct configuration then provide the listing
+	if err := s.config.EnsureValid(); err != nil {
+		updatedConfig, err := s.prompter.PromptForConfig(ctx)
+		if err != nil {
+			return []*contracts.EnvListEnvironment{}, nil
+		}
+
+		s.config = updatedConfig
+	}
+
+	matches, err := s.matchingEnvironments(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get devcenter environment list: %w", err)
+	}
+
+	envListEnvs := []*contracts.EnvListEnvironment{}
+	for _, environment := range matches {
+		envListEnvs = append(envListEnvs, &contracts.EnvListEnvironment{
+			Name:       environment.Name,
+			DotEnvPath: environment.ResourceGroupId,
+		})
+	}
+
+	return envListEnvs, nil
+}
+
+// Get returns the environment for the given name
+func (s *EnvironmentStore) Get(ctx context.Context, name string) (*environment.Environment, error) {
+	// If the devcenter configuration is not valid then we don't have enough information to query for the environment
+	if err := s.config.EnsureValid(); err != nil {
+		return nil, fmt.Errorf("%s %w", name, environment.ErrNotFound)
```

Comment:
Is it ever only possible to get `err == environment.ErrNotFound` ?   Or should we check error.is?

Reply:
If you don't have a valid dev center configuration we can't really find your environment.  We could surface some devcenter specific error.  Will think on it.